bokeh/charts/builder/bubble_builder.py

```diff
@@ -0,0 +1,151 @@
+#-----------------------------------------------------------------------------
+from six import string_types
+import numpy as np
+
+try:
+    import pandas as pd
+except:
+    pd = None
+
+from ..utils import chunk, cycle_colors, make_scatter
+from .._builder import create_and_build
+from ...models import ColumnDataSource, Range1d
+from ...properties import Any, Float
+from .scatter_builder import ScatterBuilder
+
+#-----------------------------------------------------------------------------
+# Classes and functions
+#-----------------------------------------------------------------------------
+
+
+def Bubble(values, sizes, **kws):
+    """ Create a Bubble chart using :class:`BubbleBuilder <bokeh.charts.builder.bubble_builder.BubbleBuilder>`
+    to render the geometry from values.
+
+    Args:
+        values (iterable): iterable 2d representing the data series
+            values matrix.
+        sizes (iterable(str|1d iterable)|str|1d iterable): can be used to specify a
+            a series of any sort that will be used to define the bubbles
+            size. It can be specified as an **1d iterable** with values
+            representing the bubbles size or a **string**|**int** that
+            corresponds to the key of the mapping on series of values
+            to be used for sizes and not x, y position [if
+            area.values is a mapping (like a dict, an OrderedDict
+            or a pandas DataFrame)]
+
+    In addition the the parameters specific to this chart,
+    :ref:`charts_generic_arguments` are also accepted as keyword parameters.
+
+    Returns:
+        a new :class:`Chart <bokeh.charts.Chart>`
+
+    Examples:
+
+    .. bokeh-plot::
+        :source-position: above
+
+        from collections import OrderedDict
+        from bokeh.charts import Bubble
+        from bokeh.plotting import output_file, show
+
+        # (dict, OrderedDict, lists, arrays and DataFrames of (x, y) tuples are valid inputs)
+        xyvalues = OrderedDict()
+        xyvalues['python'] = [(1, 2), (3, 3), (4, 7), (5, 5), (8, 26)]
+        xyvalues['pypy'] = [(1, 12), (2, 23), (4, 47), (5, 15), (8, 46)]
+        xyvalues['jython'] = [(1, 22), (2, 43), (4, 10), (6, 25), (8, 26)]
+        output_file('bubbles.html')
+        bubbles = Bubble(xyvalues, sizes=[[1,4,3,6,7], [5,5,2,1,5], [4,2,5,8,8]]
+        title="Bubbles", legend="top_left", ylabel='Languages')
+        show(bubbles)
+
+    """
+    return create_and_build(BubbleBuilder, values, sizes=sizes, **kws)
+
+
+class BubbleBuilder(ScatterBuilder):
+    """This is the Bubble class and it is in charge of plotting
+    Bubble charts in an easy and intuitive way.
+
+    Essentially, we provide a way to ingest the data, make the proper
+    calculations and push the references into a source object.
+    We additionally make calculations for the ranges. And finally add
+    the needed glyphs (markers) taking the references from the source.
+
+    # """
+    sizes = Any("circle", help="""
```

Comment:
A docstring above says that `sizes` can be `iterable(str|1d iterable)|str|1d iterable`, so we should try to encode this knowledge here. My first approximation is `Either(Seq(Either(String, Seq(Any))), Seq(Any), String)`, which should be later reduced to more readable `Seq(String | Seq(Any)) | Seq(Any) | String` (this is already allowed, but you would have to write `Seq(String() | Seq(Any())) | ...`. All this still uses `Any`, which I think is accurate, because this should allow categorical labels which, I presume, can be anything (although in practice can be strings and, which some issues, numbers). I'm complaining, because I wanted to directly translate charts to scala and, with all those `Any`s scattered around, it's not possible.